src/editor/components/modals/file-modal/manage-file-sources/file-source-list.jsx

```diff
@@ -0,0 +1,278 @@
+import React, { useState, useEffect } from "react";
+import styled from "react-emotion";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+
+import format from "date-fns/format";
+import CircularProgress from "@material-ui/core/CircularProgress";
+
+import DeleteModal from "../../../../../server/components/delete-modal";
+import {
+  deleteFileSource as deleteFileSourceAction,
+  createFileUpdateOperation as createFileUpdateOperationAction,
+  getFileSources as getFileSourcesAction
+} from "../../../../actions/file-source-actions";
+
+import {
+  TextButton,
+  OutlineButton
+} from "../../../../../shared/components/buttons";
+
+import {
+  List,
+  ListItem,
+  ListMain,
+  ListPrimaryText,
+  ListSecondaryText,
+  ListSecondaryTextLink,
+  ListMetadata
+} from "../../../../../server/components/list";
```

Comment:
should refactor all this to be in `shared`?

Reply:
yeah, probably time to do that